src/methods.jl

```diff
@@ -120,3 +120,54 @@ julia> flat_correct(frame, flat)
 * [`flat_correct!`](@ref)
 """
 flat_correct(frame::AbstractArray, flat_frame::AbstractArray; kwargs...) = flat_correct!(deepcopy(frame), flat_frame; kwargs...)
+
+
+"""
+    trim(frame::AbstractArray, idxs)
+
+Trims the frame to remove the region specified by idxs.
+
+This function trims the array in a manner such that final array should be rectangular.
+The indices follow standard Julia convention, so `(:, 45:60)` trims all columns from 45 to 60 and `(1:20, :)` trims all the rows from 1 to 20.
+
+# Examples
+```jldoctest
+julia> frame = ones(5, 5);
+
+julia> trim(frame, (:, 2:5))
+5×1 Array{Float64,2}:
+ 1.0
+ 1.0
+ 1.0
+ 1.0
+ 1.0
+
+```
+
+# See Also
+[`trim!`](@ref)
+"""
+trim(frame::AbstractArray, idxs) = collect(trim!(frame, idxs))
+
+
+"""
+    trim!(frame::AbstractArray, idxs)
```

Comment:
I'm confused by the use of `!` here: this function doesn't really modify anything, does it?

Reply:
This function is not mutating in the subroutine, but when the returned array is modified the parent array is also modified, that's why I used `!`. What's your thoughts on the use of `!`?

Reply:
Let's first make it clear that the `!` is purely a convention, nothing fundamental.  [`view`](https://docs.julialang.org/en/v1/base/arrays/#Base.view) behaves exactly like your `trim!` and doesn't have a bang in its name.

To be honest, I fail to see what these trimming functions do that basic indexing in Julia already doesn't.

Reply:
> To be honest, I fail to see what these trimming functions do that basic indexing in Julia already doesn't.

Sorry, yeah, the difference is that `trim` does the complement of indexing

Reply:
Yeah I thought about this for a bit- I would love for there to be a version that returns a view because it will be faster if it’s a piece within a pipeline. However I don’t think view should be the default, since an average wandering astronomer will inevitably overwrite their data on accident. The bang method was all I could think at the time to differentiate the two, how else could we differentiate the two?

Reply:
Yes, I honestly can't offer a better alternative which isn't too verbose.  However, note that while being faster than the `trim` function, this `trim!` allocates quite a bit:
```julia
julia> A = rand(4096, 4096);

julia> @benchmark trim!($A, $((:, 3000:4096)))
BenchmarkTools.Trial: 
  memory estimate:  804.89 KiB
  allocs estimate:  22991
  --------------
  minimum time:     1.184 ms (0.00% GC)
  median time:      1.430 ms (0.00% GC)
  mean time:        1.426 ms (5.04% GC)
  maximum time:     14.958 ms (90.78% GC)
  --------------
  samples:          3501
  evals/sample:     1

julia> @benchmark view($A, $(:), $(1:2999))
BenchmarkTools.Trial: 
  memory estimate:  64 bytes
  allocs estimate:  1
  --------------
  minimum time:     7.858 ns (0.00% GC)
  median time:      25.282 ns (0.00% GC)
  mean time:        27.180 ns (19.60% GC)
  maximum time:     13.698 μs (99.88% GC)
  --------------
  samples:          10000
  evals/sample:     999
```

Reply:
I think these allocations are because of `Not` of `InvertedIndices` and the inner design of `InvertedIndices`. Should we switch to some if-else conditions for `trim!`? this would lead to fewer allocations but the code loses its beauty of being short and succinct

Reply:
I’m surprised about the number of allocations, the underlying code that parses the indices shouldn’t be allocating on the surface. 


For the naming, let’s use `trim` and `trimview`. This is the notation used by a few of the JuliaImages functions (warpedview)

Reply:
So, should we use `InvertedIndices` with more allocations and succinct code or switch to if-else conditions and use `@view`?

Reply:
Removed using `InvertedIndices` and switched to using `setdiff` for generating complement indices, this reduced allocations by a factor of 4.